File: backend/src/main/java/com/back/domain/notification/repository/NotificationRepository.java
package com.back.domain.notification.repository;

import com.back.domain.member.entity.Member;
import com.back.domain.notification.entity.Notification;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface NotificationRepository extends JpaRepository<Notification, Integer> {
    List<Notification> findByMemberOrderByCreatedAtDesc(Member member);

    Optional<Notification> findByIdAndMember(Long id, Member member);
}

File: backend/src/main/java/com/back/domain/notification/repository/NotificationRepository.kt
package com.back.domain.notification.repository

import com.back.domain.member.entity.Member
import com.back.domain.notification.entity.Notification
import org.springframework.data.jpa.repository.JpaRepository

interface NotificationRepository : JpaRepository<Notification, Long> {
    fun findByMemberOrderByCreatedAtDesc(member: Member): List<Notification>

    fun findByIdAndMember(id: Long, member: Member): Notification?

    fun findByMember(member: Member): List<Notification>
}
